PSI Food waste/Pages/Forms/RestaurantList.cshtml

```diff
@@ -44,6 +44,7 @@
             {
                 if (@Model._registerRepository.CurrentUser.SubscribedRestaurants.Contains(restaurant))
```

Comment:
Here, even if restaurant is in the list, the subscribe button still appears. On top of that, in the page model, it adds restaurant duplicates, as if the check "contains" is no longer working

Reply:
Not really sure, I first thought it was that when a restaurants gets added to the data base the primary key gets assigned to it, so the restaurant gets added to the current user list without a id. But it turns out wrong

Reply:
I have done some testing and it seems that the compare is broken. Even if restaurants are the same, contains function still returns false. "Equals" returns same results, always false, even if objects have the same values. The only thing that worked for me right now is matching restaurant IDs. I might make it as a temporary solution

Reply:
ok, so it was checking by reference, no clue how and why it was working before. I've overridden the Equals functions and now it works as expected.